corgidrp/polflat.py

```diff
@@ -0,0 +1,377 @@
+import numpy as np
+from scipy import interpolate
+from scipy.ndimage import median_filter
+from scipy.ndimage import gaussian_filter as gauss
+from scipy import ndimage
+from scipy.signal import convolve2d
+import astropy.io.fits as fits
+from astropy.convolution import convolve_fft
+import photutils.centroids as centr
+from photutils.aperture import CircularAperture
+import corgidrp.mocks as mocks
+import corgidrp.data as data
+from astropy.stats import sigma_clipped_stats
+from photutils.detection import DAOStarFinder
+import corgidrp.flat as flat
+from skimage.transform import downscale_local_mean
+
+                                     
+def create_spatial_pol(dataset,nr=60,pfov_size=174,image_center_x=512,image_center_y=512,separation_diameter_arcsec=7.5,alignment_angle_WP1=0,alignment_angle_WP2=45,planet=None,band=None,prism='POL0'):
+    """Turns a dataset of neptune or uranus images with single planet images into the images observed through the wollaston prisms also incorporates the spatial variation of polarization on the 
+        surface of the planet
+
+    
+        Args:
+            dataset (corgidrp.data.Dataset): a dataset of image frames that are raster scanned (L2a-level)
+            nr (int): planet radius
```

Comment:
should nr default to None as the value? since you have code further down in lines 73-78 to guess the nr based on the target name.